lib/AST/ASTScopeCreation.cpp

```diff
@@ -0,0 +1,1061 @@
+  dispatchAndCreateIfNeeded(stmt->getBody(), deferred);
+  ScopeCreator().visitExpr(stmt->getCond(), this, deferred);
+}
+
+void DoCatchStmtScope::expandMe(DeferredNodes &deferred) {
+  dispatchAndCreateIfNeeded(stmt->getBody(), deferred);
+
+  for (auto catchClause : stmt->getCatches()) {
+    ScopeCreator().visitCatchStmt(catchClause, this, deferred);
+  }
+}
+
+void SwitchStmtScope::expandMe(DeferredNodes &deferred) {
+  ScopeCreator().visitExpr(stmt->getSubjectExpr(), this, deferred);
+
+  for (auto caseStmt : stmt->getCases())
+    createSubtree<CaseStmtScope>(caseStmt);
+}
+
+void ForEachStmtScope::expandMe(DeferredNodes &deferred) {
+  ScopeCreator().visitExpr(stmt->getSequence(), this, deferred);
+
+  // Add a child describing the scope of the pattern.
+  createSubtree<ForEachPatternScope>(stmt);
+}
+
+void ForEachPatternScope::expandMe(DeferredNodes &deferred) {
+  ScopeCreator().visitExpr(stmt->getWhere(), this, deferred);
+  ScopeCreator().visitBraceStmt(stmt->getBody(), this, deferred);
+}
+
+void CatchStmtScope::expandMe(DeferredNodes &deferred) {
+  ScopeCreator().visitExpr(stmt->getGuardExpr(), this, deferred);
+  dispatchAndCreateIfNeeded(stmt->getBody(), deferred);
+}
+
+void CaseStmtScope::expandMe(DeferredNodes &deferred) {
+  for (auto &caseItem : stmt->getMutableCaseLabelItems())
+    ScopeCreator().visitExpr(caseItem.getGuardExpr(), this, deferred);
+
+  // Add a child for the case body.
+  dispatchAndCreateIfNeeded(stmt->getBody(), deferred);
+}
+
+void GuardStmtScope::expandMe(DeferredNodes &deferred) {
+  // Add a child to describe the guard condition.
+  createSubtree<GuardConditionalClauseScope>(stmt, 0);
+
+  // Add a child for the 'guard' body, which always exits.
+  dispatchAndCreateIfNeeded(stmt->getBody(), deferred);
+
+  // Add a child to describe the guard condition for the continuation.
+  createSubtreeWithDeferrals<GuardContinuationScope>(deferred, stmt, 0);
+}
+
+void BraceStmtScope::expandMe(DeferredNodes &deferred) {
+  deferred.pushAllNecessaryNodes(stmt->getElements());
+  dispatchAndCreateAll(deferred);
+}
+
+void VarDeclScope::expandMe(DeferredNodes &deferred) {
+  assert(deferred.empty() && "Decls needing this var go into the PatternEntryUseScope, not here");
+  addChildrenForAllExplicitAccessors(decl, deferred);
+}
+
+void SubscriptDeclScope::expandMe(DeferredNodes &deferred) {
+  auto *sub = decl;
+  auto *leaf = createGenericParamScopes(sub, sub->getGenericParams());
+  auto *params = leaf->createSubtree<AbstractFunctionParamsScope>(
+      sub->getIndices(), sub->getGetter());
+  params->addChildrenForAllExplicitAccessors(sub, deferred);
+}
+
+void WholeClosureScope::expandMe(DeferredNodes &) {
+  if (auto *cl = captureList.getPtrOrNull())
+    createSubtree<CaptureListScope>(cl);
+  if (closureExpr->getInLoc().isValid())
+    createSubtree<ClosureParametersScope>(captureList, closureExpr);
+  createSubtree<ClosureBodyScope>(captureList, closureExpr);
+}
+
+
```

Comment:
Another inconsistency around newlines.

Reply:
Yup, fixed that one. But there will be places where I think that an extra newline can increase readability, just not there! Thanks.